python/services/evaluationservice/dmod/evaluationservice/evaluation_service/consumers/listener.py

```diff
@@ -565,19 +606,34 @@ async def subscribe_to_channel(self, payload: typing.Dict[str, typing.Any] = Non
                 request_id=payload.get(REQUEST_ID_KEY)
             )
 
-    async def receive(self, text_data=None, **kwargs):
+    async def receive(self, text_data: str = None, bytes_data: bytes = None, **kwargs):
         """
         Processes messages received via the socket.
 
         Called when the other end of the socket sends a message
 
         Args:
-            text_data: The data sent over the socket
-            **kwargs:
+            text_data: Data sent over the socket in the form of text data
+            bytes_data: Data sent over the socket in the form of binary data
         """
+        request_id: typing.Optional[str] = kwargs.get(REQUEST_ID_KEY)
+
+        if not text_data and bytes_data:
+            try:
+                text_data = bytes_data.decode("utf-8")
```

Comment:
Not a problem, but it caught my eye that you went out of your way to be explicit about the encoding.  I'm a bit curious if there was any particular reason behind it.  Also, might it be worth adding an `encoding` param to this function, with a default of `"utf-8"`, and using it here?

Reply:
I have several different autocomplete tools competeting for attention since everything has to be driven by some version of AI nowadays and that's just what the tab key provided in that instance, lol. It's not technically wrong (it's the default), so I let it be. Want me to remove it to remove the source of confusion?

Reply:
No, I don't think there's any need if that's all that was behind this.